### PR TITLE
fix(propagator-ot-trace): read sampled flag correctly from span context

### PR DESCRIPTION
**Pull request by basti1302**
## Which problem is this PR solving?

Fixes the way the `sampled` flag is read from the span context. The way it
is currently read will break when more flags are introduced. This is even
mentioned explicitly in the
[specification](https://www.w3.org/TR/trace-context/#trace-flags).

## Short description of the changes

- check the flag bit independently instead of comparing the whole flags byte value with the SAMPLED_FLAG constant.
- add a unit test that show cases the issue (and breaks without the change in production code)

## Checklist

- [x] Ran `npm run test` for the edited package(s) on the latest commit if applicable.

### CODE REVIEW

**Comment by basti1302** [comment]
For the record, I am currently working with the CCLA managers at my org to get my CLA in place.

**Review by dyladan — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

**Comment by basti1302** [comment]
I have the CLA in place now. It shows up okay on https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1081, but not here. If I go through the EasyCLA process again for this PR (clicking on the links in the EasyCLA bot message in this PR), it ends with this

image

... but after that, the check still shows up as failed.

I guess EasyCLA does not handle multiple simultaneous PRs from a first time contributor all that well? Should I just close this PR and open it again? Or use the support ticket link? ("For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

**Comment by Flarna** [comment]
@basti1302 Try to push a commit, maybe this retriggers the check here.

**Comment by basti1302** [comment]
> @basti1302 Try to push a commit, maybe this retriggers the check here.

It did! Thanks 👍   CLA shows up as "covered" now.

**Review by blumamir — APPROVED** [review]

**Comment by basti1302** [comment]
I think this change is ready to be merged.

**Comment by Flarna** [comment]
@basti1302 it requires a rebase/merge from main.
I have no rights to do this on your fork, most likely because its owned by an org. In that case the "allow maintainer edits" doesn't work.

**Comment by basti1302** [comment]
> @basti1302 it requires a rebase/merge from main. I have no rights to do this on your fork, most likely because its owned by an org. In that case the "allow maintainer edits" doesn't work.

I had rebased this branch earlier today, but of course after the merge of https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1078 it now needed yet another rebase, which I have performed just now.

**Pull request merged**
